config/functions/common-functions

```diff
@@ -54,54 +54,126 @@ time_cal() {
 ## Install essential packages
 prepare_host() {
```

Comment:
we should expose this its own .sh and `RUN add ./prepare_host.sh` in the Dockerfile to prevent redundantly having to write the dependency here and in the Dockerfile.